src/main/java/com/amazon/ion/impl/bin/IonRawBinaryWriter_1_1.kt

```diff
@@ -448,6 +467,9 @@ class IonRawBinaryWriter_1_1 internal constructor(
                 List, SExp, Struct -> {
                     val contentLength = currentContainer.length
                     if (contentLength <= 0xF && !currentContainer.usesFlexSym) {
+                        // TODO: Right now, this is skipped if we switch to FlexSym after starting a struct
```

Comment:
🗺️ This is incidental. I was aware of this when I implemented structs, but I had since forgotten it, and then I rediscovered this while I was writing tests for E-Exps.

Reply:
The scenario is when we have a struct that starts out as a SID struct, has at least one SID field before switching to FlexSyms, and has a total length of less than 16 bytes. The consequence of this limitation is that we're wasting a byte. E.g.:
```
FC             | Variable length SID Struct     ]____ Could be replaced with
17             | Length = 11                    ]     just `CB` to save a byte.
81             | SID 64
5E             | true
01             | switch to FlexSym encoding
FB 66 6F 6F    | FlexSym 'foo'
5E             | true
02 01          | FlexSym SID 64
5E             | true
```
Ideally, we could write this starting with `CB` but we don't store enough state right now to track whether it's safe to do that. We only know that the struct is _currently_ using FlexSyms, which means that we don't know whether the OpCode at the start is `FC` or `FD`. If it's `FC`, then it's safe to collapse `FC 17` to `CB` (as in the example above). If it actually started with `FD`, then going to `CB` would cause the struct to be malformed.

It's not difficult to solve this—it just requires a way to check how the struct started out and a few more branches for the different scenarios—but I didn't really want to address it while the struct encoding is still in a bit of flux.